pkg/mdm/mdmtest/apple.go

```diff
@@ -199,6 +228,119 @@ func (c *TestAppleMDMClient) fetchEnrollmentProfileFromDEPURL() error {
 	)
 }
 
+func (c *TestAppleMDMClient) fetchEnrollmentProfileFromOTAURL() error {
+	rawDeviceInfo := []byte(fmt.Sprintf(`<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>PRODUCT</key>
+	<string>%s</string>
+	<key>SERIAL</key>
+	<string>%s</string>
+	<key>UDID</key>
+	<string>%s</string>
+	<key>VERSION</key>
+	<string>22A5316k</string>
+</dict>
+</plist>`, c.Model, c.SerialNumber, c.UUID))
+
+	do := func(cert *x509.Certificate, key *rsa.PrivateKey) ([]byte, error) {
+		signedData, err := pkcs7.NewSignedData(rawDeviceInfo)
+		if err != nil {
+			return nil, fmt.Errorf("create signed data: %w", err)
+		}
+		err = signedData.AddSigner(cert, key, pkcs7.SignerInfoConfig{})
+		if err != nil {
+			return nil, fmt.Errorf("add signer: %w", err)
+		}
+		sig, err := signedData.Finish()
+		if err != nil {
+			return nil, fmt.Errorf("finish signing: %w", err)
+		}
+
+		request, err := http.NewRequest(
+			"POST",
+			c.fleetServerURL+"/api/latest/fleet/ota_enrollment?enroll_secret="+c.otaEnrollSecret,
+			bytes.NewReader(sig),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("create request: %w", err)
+		}
+		// #nosec (this client is used for testing only)
+		cc := fleethttp.NewClient(fleethttp.WithTLSClientConfig(&tls.Config{
+			InsecureSkipVerify: true,
+		}))
+		response, err := cc.Do(request)
+		if err != nil {
+			return nil, fmt.Errorf("send request: %w", err)
+		}
+		if response.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("request error: %d, %s", response.StatusCode, response.Status)
+		}
+
+		body, err := io.ReadAll(response.Body)
+		if err != nil {
+			return nil, fmt.Errorf("read body: %w", err)
+		}
+
+		return body, nil
+	}
+
+	// TODO(roberto 09-10-2024): the first request in the OTA flow must be
+	// signed using a keypair that has a valid Apple certificate as root. I
+	// believe this could be done with a little bit of reverse
+	// engineering/cleverness but for now, we're signing the request with
+	// our mock certs and setting this env var to skip the verification.
+	os.Setenv("FLEET_DEV_MDM_APPLE_DISABLE_DEVICE_INFO_CERT_VERIFY", "1")
```

Comment:
Just to make sure, this will work for integration tests as the mocked devices and the server are in the same process, but this won't work for load tests /osquery_perf, right?

Reply:
I think you're right. For osquery_perf you'll need to add it in the terraform file that starts the fleet server. I just created a separate PR with the documentation here https://github.com/fleetdm/fleet/pull/21950 to alleviate the amount of codeowners